### ISSUE TITLE
Failed to run majority examples on Windows Bash

### ISSUE THREAD

**Issue by QhelDIV** [taichi-dev/taichi#110]
I have installed taichi on Windows Bash, with Xming ( a kind of X server for windows ) and in order to solve the "NoSuchDisplayException" rised by pyglet, I installed a dummy screen adapter ( by execute sudo apt-get install xserver-xorg-video-dummy),  
Then I am able to run a few examples successfully (e.g. rendering/trees.py ,  simulation/3d/mgpcg_smoke_3d.py), but fail to run the rest...
This 'camera ... deprecated' warning always shows up, so I suspect this warning has something to do with my failures... 
**So how can I eliminate this warning?**

Also, the typical error I encountered when running examples is **" [export_math.cpp:array2d_to_ndarray@98]"** 
how to solve this problem? Thanks in advance!
The typical error info when run an example:
```
root@QhelDIV:/home/root/repos/taichi/python/examples/rendering# python3 fractals.py
[T 01/18/18 20:51:04.217] [logging.cpp:Logger@66] Taichi core started. Thread ID = 1907
Scene loaded. Triangle count: 22
stage 1
time: 0.018679141998291016
[E 01/18/18 20:51:05.383] [export_math.cpp:array2d_to_ndarray@98]
[E 01/18/18 20:51:05.384] Received signal 6 (Aborted)
***************************
* Taichi Core Stack Trace *
***************************
/home/root/repos/taichi/build/taichi_core.so: taichi::signal_handler(int)
/lib/x86_64-linux-gnu/libc.so.6:
/lib/x86_64-linux-gnu/libpthread.so.0: raise
/home/root/repos/taichi/build/taichi_core.so: void taichi::array2d_to_ndarray<taichi::ArrayND<2, taichi::VectorND<3, float, (taichi::InstSetExt)3> >, 3>(taichi::ArrayND<2, taichi::VectorND<3, float, (taichi::InstSetExt)3> >*, unsigned long)
/home/root/repos/taichi/build/taichi_core.so:
/home/root/repos/taichi/build/taichi_core.so:
python3: PyCFunction_Call
python3: PyEval_EvalFrameEx
python3: ) [0x540199]
python3: PyEval_EvalFrameEx
python3: PyEval_EvalFrameEx
python3: PyEval_EvalFrameEx
python3: ) [0x540199]
python3: PyEval_EvalFrameEx
python3: ) [0x540199]
python3: PyEval_EvalCode
python3: ) [0x60c272]
python3: PyRun_FileExFlags
python3: PyRun_SimpleFileExFlags
python3: Py_Main
python3: main
/lib/x86_64-linux-gnu/libc.so.6: __libc_start_main
python3: _start
```

**Comment by yuanming-hu** [comment]
Thanks for the feedback. I'll deal with it after the SIGGRAPH deadline.

**Comment by yuanming-hu** [comment]
Hi QhelDIV,

I think most of them are fixed now (except those mentioned in #116). Sorry for the delay - I just recovered from the SIGGRAPH deadline.

Yuanming

**Comment by QhelDIV** [comment]
I can run most rendering examples now. ( but I still have trouble to run many simulation example.)
Thank you very much XD
